src/Transport.ts

```diff
@@ -331,320 +338,383 @@ export default class Transport {
+                  context: meta.context
+                })
               }
 
-              continue
+              // retry logic
+              if (meta.attempts < maxRetries) {
+                meta.attempts++
+                debug(`Retrying request, there are still ${maxRetries - meta.attempts} attempts`, params)
+
+                // exponential backoff on retries, with jitter
+                const backoff = options.retryBackoff ?? this[kRetryBackoff]
+                const backoffWait = backoff(0, 4, meta.attempts)
+                if (backoffWait > 0) {
+                  await setTimeoutPromise(backoffWait * 1000)
+                }
+
+                continue
+              }
+
+              // Wrap the error to get a clean stack trace
+              const wrappedError = error.name === 'TimeoutError'
+                ? new TimeoutError(error.message, result, errorOptions)
+                : new ConnectionError(error.message, result, errorOptions)
+              this[kDiagnostic].emit('response', wrappedError, result)
+              throw wrappedError
             }
 
-            // Wrap the error to get a clean stack trace
-            const wrappedError = error.name === 'TimeoutError'
-              ? new TimeoutError(error.message, result, errorOptions)
-              : new ConnectionError(error.message, result, errorOptions)
-            this[kDiagnostic].emit('response', wrappedError, result)
-            throw wrappedError
+            // edge cases, such as bad compression
+            default:
+              this[kDiagnostic].emit('response', error, result)
+              throw error
           }
-
-          // edge cases, such as bad compression
-          default:
-            this[kDiagnostic].emit('response', error, result)
-            throw error
         }
       }
+
+      return returnMeta ? result : result.body
     }
 
-    return returnMeta ? result : result.body
+    // wrap in OpenTelemetry span
+    if (params.meta?.name != null) {
+      return await otelTracer.startActiveSpan(params.meta.name, async (span) => {
+        otelSpan = span
+        otelSpan.setAttributes({
+          'db.system': 'elasticsearch',
+          'http.request.method': params.method,
+          'db.operation.name': params.meta?.name
+        })
+
+        if (params.meta?.pathParts != null) {
+          for (const key of Object.keys(params.meta.pathParts)) {
+            otelSpan.setAttribute(`db.elasticsearch.path_parts.${key}`, params.meta.pathParts[key])
+          }
+        }
+
+        let response
+        try {
+          response = await _request()
+        } catch (err: any) {
+          otelSpan.recordException(err as Exception)
+          otelSpan.setStatus({ code: SpanStatusCode.ERROR })
+          otelSpan.setAttribute('error.type', err.name ?? 'Error')
+
+          throw err
+        } finally {
+          otelSpan.end()
+        }
+
+        return response
+      })
+    } else {
+      return await _request()
```

Comment:
This is very much a nit, and this isn't my repo at all, so feel free to ignore. :)
If `_request` were pulled out to a top-level method on `class Transport`, then I think the diff would look **much** smaller. Mostly there is just indentation change here, but it is big enough that `git diff -w` can't fully make a small diff -- at least for me.

Reply:
I get you. I could go either way, but your way does make future git blames more useful. Creating the otel variables in a closure was my reasoning, but passing the span as an optional third arg to a private method is just as good in this case. Fixed in 1cae465a681a83935d9b566a39c3dce3acead714.